### PR TITLE
New exceptions

### PR DESCRIPTION
**Pull request by klaasnicolaas**
This PR will add:
- Exception when you hit the rate limit
- Exception when the location isn't in the data coverage zone of the API (or other errors that the API can give).

### CODE REVIEW

**Comment by klaasnicolaas** [comment]
Maybe change the NoCoverage exception to BadRequest and add debug info in documentation? 🤔

File: forecast_solar/__init__.py [hunk 1]
```diff
@@ -124,12 +151,22 @@ async def estimate(self) -> Estimate:
...
+            ForecastSolarRequestError: There is something wrong with the
+                variables used in the request.
         """
         data = await self._request(
             f"estimate/{self.latitude}/{self.longitude}"
             f"/{self.declination}/{self.azimuth}/{self.kwp}",
             params={"time": "iso8601", "damping": str(self.damping)},
         )
+
+        if not data["result"]["watts"]:
```
Comment (balloob) [inline forecast_solar/__init__.py:141]: Do we need this? If the location is not supported, we get a 400:

```
forecast_solar.exceptions.ForecastSolarRequestError: Can't find location data for latitude = 0 and longitude = 0 (error 5)
```

File: forecast_solar/__init__.py [hunk 2]
```diff
@@ -86,28 +95,22 @@ async def _request(
             self.session = ClientSession()
             self.close_session = True
 
-        try:
-            with async_timeout.timeout(30):
```
Comment (balloob) [inline forecast_solar/__init__.py:90]: No need to handle timeout. That should be decided by the code that calls us.

File: forecast_solar/__init__.py [hunk 3]
```diff
@@ -86,28 +92,22 @@ async def _request(
...
-                )
-                response.raise_for_status()
-        except asyncio.TimeoutError as exception:
-            raise ForecastSolarConnectionError(
-                "Timeout occurred while connecting to Forecast.Solar API"
-            ) from exception
-        except (
-            ClientError,
-            ClientResponseError,
-            socket.gaierror,
```
Comment (balloob) [inline forecast_solar/__init__.py:106]: I don't think aiohttp raises socket errors? It's always wrapped in some aiohttp error. It's fine to raise those as-is.

File: forecast_solar/__init__.py (hunk 1 above)
Comment (klaasnicolaas) [reply forecast_solar/__init__.py:143]: the 400 was too general in previous error reports and was not clear to the user.

Comment (klaasnicolaas) [reply forecast_solar/__init__.py:143]: Removed it, the new error exception handling is better 😉

**Pull request merged**